docs/architecture/architecture.md

```diff
@@ -0,0 +1,330 @@
+            ],
+            "connectedTo": ["card_42"]
+        }
+    ]
+}
+```
+
+Additional info about VisualRules: [DCA documentation](https://colabora.tid.es/dca/SitePages/Inicio.aspx)
+(RESTAPI-SBC_2.6, section 6.15)
+
+#### "Plain" rules
+
+A simplified format in JSON can be used to represent rules. The former visual rule would be represented so:
+
+```json
+{
+    "name": "prueba-test",
+    "text": "select * from pattern [every ev=iotEvent((cast(id?, String)  regexp  \"asd\"))]",
+    "action": {
+        "type": "email",
+        "template": "DCA message",
+        "parameters": {
+            "to": "MANDATORY",
+            "from": "dca_support@tid.es",
+            "subject": "DCA message"
+        }
+    }
+}
+```
+
+Actually, visual rules are translated to "plain" rules by perseo internally. This is transparent for the portal which
+only see visual rules (create, update, delete)
+
+#### EPL
+
+The rule at perseo-core is expressed as an EPL sentence. EPL is a domain language of [Esper](http://esper.codehaus.org),
+the engine for processing events used in perseo-core. This EPL sentence matches an incoming event if satisfies the
+conditions and generates an "action-event" that will be sent back to perseo to execute the associated action. The EPL is
+generated by perseo for visual rules and must be included in "plain" rule, following several conditions to work in the
+interaction between perseo and perseo-core.
+
+### Action
+
+The action included in rule allows sending an email, sending an SMS or updating an attribute of the entity which was the
+source of the firing event, etc. The message sent in SMS or email can be a template with parameters replaced with fields
+of the generated event when the message is about to be sent by perseo (through SMS gateway or SMTP server)
+
+#### Loop detection
+
+An infinite loop can be created by rules with update actions, each one making a change which triggers the other rule. It
+is a symptom of a bad design or an error in writing the rules. For example, two rules like
+
+    IF temperature < 5 AND alarm == OFF THEN alarm = ON
+    IF temperature < 5 AND alarm == ON THEN alarm = OFF
+
+If the subscription in Orion is for every attribute or just for `temperature` and `alarm`, those rules will create an
+infinite loop of triggers/updates.
+
+In a best-effort to avoid this situation as much as possible, the feature of propagation of the header field
+`Fiware-correlator` is used. This header field is taken from the incoming request (or created if not present) and sent
+to every external system. The Context Broker follows the same tactic. In a loop, the same correlator will be sent
+between perseo and orion, like a ping-pong game. So, if perseo-core sends an action to perseo-fe and that action (for
+the same rule) has been executed with that correlator already, it declines executing it. It could be part of an infinite
+loop. The fact is logged and the action is ignored.
+
+<a name="Dataflow"></a>
```

Comment:
Pointless anchor `data-flow` already exists below

Reply:
Ok. Fixed in ef4582b